utils/bids/download_run_level.py

```diff
@@ -0,0 +1,364 @@
+                    log.info(f"Not actually downloading it because {bids_dir} exists")
+                else:
+                    # don't filter by subject or session, grab all
+                    bids_path = gtk_context.download_project_bids(
+                        src_data=src_data, folders=folders, dry_run=dry_run
+                    )
+
+            elif run_level == "subject":
+
+                log.info(
+                    'Downloading BIDS for subject "%s"', hierarchy["subject_label"]
+                )
+
+                if Path(bids_dir).exists():
+                    bids_path = bids_dir
+                    log.info(f"Not actually downloading it because {bids_dir} exists")
+                else:
+                    # only download this subject
+                    bids_path = gtk_context.download_project_bids(
+                        src_data=src_data,
+                        folders=folders,
+                        dry_run=dry_run,
+                        subjects=[hierarchy["subject_label"]],
+                    )
+
+            elif run_level == "session":
+
+                log.info(
+                    'Downloading BIDS for session "%s"', hierarchy["session_label"]
+                )
+
+                if Path(bids_dir).exists():
+                    bids_path = bids_dir
+                    log.info(f"Not actually downloading it because {bids_dir} exists")
+                else:
+                    # only download data for this session AND this subject
+                    bids_path = gtk_context.download_project_bids(
+                        src_data=src_data,
+                        folders=folders,
+                        dry_run=dry_run,
+                        subjects=[hierarchy["subject_label"]],
+                        sessions=[hierarchy["session_label"]],
+                    )
+
+            elif run_level == "acquisition":
+
+                if hierarchy["acquisition_label"] == "unknown acquisition":
+                    msg = (
+                        'Cannot download BIDS for acquisition "'
+                        + hierarchy["acquisition_label"]
+                        + '"'
+                    )
+                    log.critical(msg)
+                    extra_tree_text += f"ERROR: {msg}\n"
+                    bids_path = None
+                    err_code = 23  # attempt to download unknown acquisition
+
+                else:
+                    log.info(
+                        'Downloading BIDS for acquisition "%s"',
+                        hierarchy["acquisition_label"],
+                    )
+
+                    bids_path = bids_dir
+                    if Path(bids_dir).exists():
+                        log.info(
+                            "Not actually downloading it because " f"{bids_dir} exists"
+                        )
+                    else:
+                        # only download acquisition data
+                        download_bids_dir(
```

Comment:
And as a reminder, download_bids_dir is not in this codebase. You likely have it installed locally which is why you aren't getting errors, but you need to include the bids-client repo in order to access that within a Docker container.

Reply:
All of the tests are run inside the docker container so it must have been installed as a dependency of something else.  But I have added it to `requirements.txt` to be explicit about the version.